keps/sig-network/1880-services-ip-ranges/README.md

```diff
@@ -0,0 +1,1015 @@
+
+Currently, the Service CIDRs are configured independently in each kube-apiserver using flags. During
+the bootstrap process, the apiserver uses the first IP of each range to create the special
+"kubernetes.default" service. It also starts a reconcile loop, that synchronize the state of the
+bitmap with the assigned IPs to the Services.
+
+With current implementation, that each kube-apiserver can boot with different ranges configured.
+There is no conflict resolution, each apiserver keep writing and deleting others apiservers allocator bitmap
+and Services.
+
+In order to be completely backwards compatible, the bootstrap process will remain the same, the
+difference is that instead of creating a bitmap based on the flags, it will create new
+ServiceIPRange object from the flags (flags configuration removal is out of scope of this KEP) 
+with a special label `networking.kubernetes.io/service-ip-cidr-default` set to true.
+
+With the new implementation only the Service IP Range configuration will be consistent, but not the
+default IP Family, that will still depend on the flags configuration of each individual apiserver.
+However, users still can provide consistency by using a webhook or admission controller for
+defaulting the IPFamily or IPFamilyPolicy fields on Services (to be revisited before Beta).
+
+There can be multiple default ServiceIPRange objects, also for the same IP family, since
+apiservers with different configuration can work, the only condition is that the defined
+service IP ranges contain the `kubernetes.default` Service.
+
+The new bootstrap process will be:
+
+```
+read_flags
+if invalid flags
+  exit
+get ServiceCIDR objects
+if empty || no default
+   create default ServiceCIDR from flags
+   run
+else
+   if default matches flags
+      run
+   else if overlaps
+      log/event
+      run
+   else
+      log/event periodically
+      run
+```
+
+[![](https://mermaid.ink/img/pako:eNp1ksFuwjAMhl_Fygkk2AP0AFpbBpvEBbYLbQ9W40KkNKnStBuq-u4LLRXZpuUQRf4__7aTdCzXnFjAzgarC7zHqQK3npNQa5vBcrmCt5kh5FCTaUVOy1xwA4XEcz0fZNh3LUrBx9i6h1SNHnsng9JuH9YKTskLWpSZJ1-pHkzC2ZYsHMcS0Wt8qOcjNe7hjf1EYZ8-lBVyTPG1ySfqOBXYSAv0JWq77j2GjNFmoE5Ti5GfG3cl2vwyzeHXj31umxwalXnCMOMKNp1uyUisfiVv_kv25LvFzmvrZyj8G3pY7JLIPZEl_wIzr9yDvA0-Dc8WrCRTouDu-btbLGX2QiWlLHDH-0WmLFW9Q5uKuwobLqw2LChQ1rRg2Fh9vKqcBdY0NEGxQPebyjvVfwMtxq4y)](https://mermaid.live/edit#pako:eNp1ksFuwjAMhl_Fygkk2AP0AFpbBpvEBbYLbQ9W40KkNKnStBuq-u4LLRXZpuUQRf4__7aTdCzXnFjAzgarC7zHqQK3npNQa5vBcrmCt5kh5FCTaUVOy1xwA4XEcz0fZNh3LUrBx9i6h1SNHnsng9JuH9YKTskLWpSZJ1-pHkzC2ZYsHMcS0Wt8qOcjNe7hjf1EYZ8-lBVyTPG1ySfqOBXYSAv0JWq77j2GjNFmoE5Ti5GfG3cl2vwyzeHXj31umxwalXnCMOMKNp1uyUisfiVv_kv25LvFzmvrZyj8G3pY7JLIPZEl_wIzr9yDvA0-Dc8WrCRTouDu-btbLGX2QiWlLHDH-0WmLFW9Q5uKuwobLqw2LChQ1rRg2Fh9vKqcBdY0NEGxQPebyjvVfwMtxq4y)
+
+#### The special "default" ServiceIPRanges
+
+There has to be always one default ServiceIPRange containing the `kubernetes.default` Service.
```

Comment:
I think this statement is misleading.  How about something like:

The `kubernetes.default` Service is expected to be covered by a valid range.  Each apiserver will ensure that a range object exists to cover its own flag-defined ranges, so this should be true in normal cases.  If someone were to force-delete the range covering `kubernetes.default` it would be treated the same as any Service in the repair loop, which will generate warnings about orphaned Service IPs.